java-security/src/main/java/com/sap/cloud/security/token/validation/validators/XsuaaJwtIssuerValidator.java

```diff
@@ -38,27 +37,37 @@
 	@Override
 	public ValidationResult validate(Token token) {
 		String tokenKeyUrl = token.getHeaderParameterAsString(TokenHeader.JWKS_URL);
+		URI jkuUri;
+
 		if (tokenKeyUrl == null || tokenKeyUrl.trim().isEmpty()) {
 			return createInvalid(
 					"Issuer validation can not be performed because Jwt token does not contain 'jku' header parameter.");
 		}
-		return matchesTokenKeyUrlDomain(tokenKeyUrl);
-	}
-
-	private ValidationResult matchesTokenKeyUrlDomain(String tokenKeyUrl) {
-		URI jkuUri;
 		try {
-			jkuUri = new URI(tokenKeyUrl);
-			if (jkuUri.getHost() != null && jkuUri.getHost().endsWith(domain)) {
-				return createValid();
-			}
-		} catch (URISyntaxException e) {
-			logger.error("Error: 'jku' header parameter '{}' does not provide a valid URI: {}.", tokenKeyUrl,
-					e.getMessage(), e);
+			jkuUri = URI.create(tokenKeyUrl);
+		} catch (IllegalArgumentException e) {
+			return createInvalid(
+					"Issuer validation can not be performed because Jwt token does not contain a valid uri as 'jku' header parameter.");
+		}
+		if (!matchesTokenKeyUrlDomain(jkuUri)) {
+			return createInvalid(
+					"Issuer is not trusted because 'jku' '{}' does not match uaa domain '{}' of the identity service.",
+					jkuUri, domain);
+		}
+		if (!matchesTokenKeyEndpoint(jkuUri)) {
+			return createInvalid(
+					"Jwt token does not contain a valid 'jku' header parameter.",
+					jkuUri, domain);
 		}
-		return createInvalid(
-				"Issuer is not trusted because 'jku' '{}' does not match uaa domain '{}' of the identity service.",
-				tokenKeyUrl, domain);
+		return createValid();
+	}
+
+	private boolean matchesTokenKeyUrlDomain(URI jkuUri) {
+		return jkuUri.getHost() != null && jkuUri.getHost().endsWith(domain);
+	}
+
+	private boolean matchesTokenKeyEndpoint(URI jkuUri) {
+		return jkuUri.getPath().endsWith("token_keys") && jkuUri.getQuery() == null && jkuUri.getFragment() == null;
```

Comment:
I think you should better do:
`.endsWith("/token_keys") `
with the slash

Reply:
yeah, but path starts not with token_keys, so that is appropriate.